### PR TITLE
🔥 🔥 Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by roniy68**
# Highlights 🥇 

## What I Have Done: 
✅ Install CanCanCan in your project.
✅ Add a role column to the users table. Remember to use a migration for this.
✅ A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
      - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
      
✅ A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
      - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### Tests : 
✅ User index page
✅ user show page
✅ User post index page
✅ Post show page

# Local Installation 

```
bundle install &&  EDITOR="code --wait" rails credentials:edit 
```

### put your db user and pass as env variable

```
POSTGRESQL_USER: username
POSTGRESQL_PASSWORD: password
RAILSBLOG_DATABASE_PASSWORD: password
```

## 👥 Authors

- Ahmed Hassan Rony @roniy68 
- Muhammad Aleem @MAleemH 

## Missed anything?
- [ ] Explained the purpose of this PR.
- [ ] Self reviewed the PR.
- [ ] Added or updated test cases.
- [ ] Informed of breaking changes, testing and migrations (if applicable).
- [ ] Updated documentation (if applicable).
- [ ] Attached screenshots (if applicable).

## Followed Rules: 
- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
-  Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-  Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
+      redirect_to new_post_comment_path(post), alert: t('.not_created')
+    end
+  end
+
+  def destroy
+    comment = Comment.find(params[:id])
+
+    if comment.destroy
+      redirect_to user_post_path(comment.author, post), notice: t('.deleted')
+      post.update_post_counter
```
Comment (Munsa1) [inline app/controllers/comments_controller.rb:24]: - [ ] As requeted by the previous reviewer in pull request #11, which you recently closed, kindly fix this error which comes up when a user tries to add a comment to another users post. 
image
## Please do not open another pull request, kindly fix the issue and ask for another review

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @roniy68, 
required-changes-2

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Munsa1) [reply app/controllers/comments_controller.rb:24]: - [ ] I almost forgot, I would encourage you to use the old PR #11 for all re-reviews from now on

Comment (roniy68) [reply app/controllers/comments_controller.rb]: 🔥 Sorry I had to debug a little longer to fix the issue. 

# ✅ Issue Solved : 

image

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the required changes requested by the Previous review👏
- The App has cancancan Gem 👏
- The App is working fine 👏


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
